### PR TITLE
fix: Windows Azure CNI conflist should split ExceptionList by IP family

### PR DESCRIPTION
**Pull request by tyler-lloyd**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

Currently writing the outboundNAT exception list for dual-stack
with both v4 and v6 CIDRs in the same list, but HNS expects them
to be in 2 different lists, one per IP family.

Going from
```
{
                    "Name": "EndpointPolicy",
                    "Value": {
                        "Type": "OutBoundNAT",
                        "ExceptionList": [
                            "10.244.0.0/16",
                            "fd12:3456::/64"
                        ]
                    }
                },
```

to

```
{
                    "Name": "EndpointPolicy",
                    "Value": {
                        "Type": "OutBoundNAT",
                        "ExceptionList": [
                            "10.244.0.0/16"
                        ]
                    }
                },
{
                    "Name": "EndpointPolicy",
                    "Value": {
                        "Type": "OutBoundNAT",
                        "ExceptionList": [
                            "fd12:3456::/64"
                        ]
                    }
                },
```

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [x] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Review by ashvindeodhar — APPROVED** [review]

File: staging/cse/windows/azurecnifunc.ps1 [hunk 1]
```diff
@@ -110,7 +110,35 @@ function Set-AzureCNIConfig
...
+                # we just assume the first entry in additional Args is the exception
+                # list for IPv4 and then append a new EnpointPolicy for IPv6. We
+                # probably shouldn't hard code the first one like this and just build
+                # 2 EndpointPolicies and append to the AdditionalArgs.
+                $configJson.plugins.AdditionalArgs[0].Value.ExceptionList = $ipv4Cidrs
+
+                $outboundException = [PSCustomObject]@{
+                    Name = 'EndpointPolicy'
+                    Value = [PSCustomObject]@{
+                        Type = 'OutboundNAT'
```
Comment (ashvindeodhar) [inline staging/cse/windows/azurecnifunc.ps1:134]: this needs to be OutBoundNAT

**Review by ashvindeodhar — CHANGES_REQUESTED** [review]

File: staging/cse/windows/azurecnifunc.ps1 (hunk 1 above)
Comment (tyler-lloyd) [reply staging/cse/windows/azurecnifunc.ps1]: fixed

File: staging/cse/windows/azurecnifunc.ps1 [hunk 2]
```diff
@@ -110,7 +110,35 @@ function Set-AzureCNIConfig
             $configJson.plugins.AdditionalArgs[0].Value.ExceptionList = $processedExceptions
         }
         else {
-            $configJson.plugins.AdditionalArgs[0].Value.ExceptionList = $exceptionAddresses
+            if ($IsDualStackEnabled) {
```
Comment (AbelHu) [inline staging/cse/windows/azurecnifunc.ps1:113]: Does IsDualStackEnabled only support WS2022? The else in line 112 is for `if ($osBuildNumber -le 17763)`.

Comment (tyler-lloyd) [reply staging/cse/windows/azurecnifunc.ps1:113]: yes, there is no plan for IPv6 support to come to WS2019

**Review by AbelHu — APPROVED** [review]
Please test this change with/without ipv6

**Pull request merged**